test/phoenix_html/link_test.exs

```diff
@@ -67,6 +67,23 @@ defmodule Phoenix.HTML.LinkTest do
            ~s[</form>]
   end
 
+  test "button with do" do
+    csrf_token = Plug.CSRFProtection.get_csrf_token()
+
+    assert safe_to_string(button(to: "/world", class: "small", do: "<span>Hi</span>")) ==
```

Comment:
This should be `raw("<span>Hi</span>") or `{:safe, ...}` then. We also need a test this will escape unsafe output.